scrapy_autoextract/middlewares.py

```diff
@@ -145,12 +145,25 @@ def process_response(self, request, response, spider):
         url = request.meta[AUTOEXTRACT_META_KEY]['original_url']
 
         try:
-            result = json.loads(response.body.decode('utf8'))[0]
+            response_object = json.loads(response.body.decode('utf8'))
         except Exception:
             self.inc_metric('autoextract/errors/json_decode')
             raise AutoExtractError('Cannot parse JSON response from AutoExtract'
                                    ' for {}: {}'.format(url, response.body))
 
+        if response.status != 200:
+            self.inc_metric('autoextract/errors/response_error')
+            raise AutoExtractError('Received error from AutoExtract for '
+                                   '{}: {}'.format(url, response_object))
```

Comment:
Based on https://github.com/scrapinghub/scrapinghub-autoextract, it is super helpful to see response body in case of error, at least as a debug message; also, I think that'd be good to include response status to metrics as well - add 
`self.inc_metric('autoextract/errors/response_error/{}'.format(response.status))`

Reply:
I'm adding the response code now, it's a good idea.
About the response body, it is already included absolutely everytime we `raise AutoExtractError`:

- https://github.com/scrapinghub/scrapy-autoextract/blob/d835936ca5d4760778160951c691d5829761f60a/scrapy_autoextract/middlewares.py#L151-L152
- https://github.com/scrapinghub/scrapy-autoextract/blob/d835936ca5d4760778160951c691d5829761f60a/scrapy_autoextract/middlewares.py#L156-L157
- https://github.com/scrapinghub/scrapy-autoextract/blob/d835936ca5d4760778160951c691d5829761f60a/scrapy_autoextract/middlewares.py#L164-L165
- https://github.com/scrapinghub/scrapy-autoextract/blob/d835936ca5d4760778160951c691d5829761f60a/scrapy_autoextract/middlewares.py#L169-L170

Which means all the errors will be logged. You think it makes sense to log them separately again, just the response body?

Reply:
About logging of response body: neither str(response_object) nor repr(response_object) contains response body, so it is not always logged.

Reply:
I see. I just pushed a commit and I looked at what's included in the ApiError in scrapinghub-autoextract, so I'm logging what's available here.

Reply:
Thanks @croqaz!